tracer/src/Datadog.Trace/AppSec/Waf/Encoder.cs

```diff
@@ -302,5 +208,391 @@ private static StringBuilder FormatList(IEnumerable<object> objs, StringBuilder
+                    break;
+                }
+
+                case IEnumerable<KeyValuePair<string, JToken>> objDict:
+                {
+                    var count = objDict is ICollection<KeyValuePair<string, JToken>> dct ? dct.Count : objDict.Count();
+                    ddwafObjectStruct = ProcessKeyValuePairs(objDict, count, &GetKey3, &GetValue3);
+                    static string GetKey3(KeyValuePair<string, JToken> item) => item.Key;
+                    static object GetValue3(KeyValuePair<string, JToken> item) => item.Value;
+                    break;
+                }
+
+                case IEnumerable<KeyValuePair<string, string[]>> objDict:
+                {
+                    var count = objDict is ICollection<KeyValuePair<string, string[]>> dct ? dct.Count : objDict.Count();
+                    ddwafObjectStruct = ProcessKeyValuePairs(objDict, count, &GetKey4, &GetValue4);
+                    static string GetKey4(KeyValuePair<string, string[]> item) => item.Key;
+                    static object GetValue4(KeyValuePair<string, string[]> item) => item.Value;
+                    break;
+                }
+
+                case IEnumerable<KeyValuePair<string, List<string>>> objDict:
+                {
+                    var count = objDict is ICollection<KeyValuePair<string, List<string>>> dct ? dct.Count : objDict.Count();
+                    ddwafObjectStruct = ProcessKeyValuePairs(objDict, count, &GetKey5, &GetValue5);
+                    static string GetKey5(KeyValuePair<string, List<string>> item) => item.Key;
+                    static object GetValue5(KeyValuePair<string, List<string>> item) => item.Value;
+                    break;
+                }
+
+                case IEnumerable enumerable:
+                {
+                    ddwafObjectStruct = new DdwafObjectStruct { Type = DDWAF_OBJ_TYPE.DDWAF_OBJ_ARRAY };
+                    if (!string.IsNullOrEmpty(key))
+                    {
+                        FillParamName(ref ddwafObjectStruct, key!);
+                    }
+
+                    if (applySafetyLimits && remainingDepth-- <= 0)
+                    {
+                        Log.Warning("EncodeList: object graph too deep, truncating nesting {Items}", string.Join(", ", enumerable));
+                        break;
+                    }
+
+                    if (enumerable is IList { Count: { } count } listInstance)
+                    {
+                        if (count > WafConstants.MaxContainerSize)
+                        {
+                            Log.Warning<int>("EncodeList: list too long, it will be truncated, MaxMapOrArrayLength {MaxMapOrArrayLength}", WafConstants.MaxContainerSize);
+                            break;
+                        }
+
+                        var childrenCount = count < WafConstants.MaxContainerSize ? count : WafConstants.MaxContainerSize;
```

Comment:
```suggestion
                        if (applySafetyLimits && count > WafConstants.MaxContainerSize)
                        {
                            Log.Warning<int>("EncodeList: list too long, it will be truncated, MaxMapOrArrayLength {MaxMapOrArrayLength}", WafConstants.MaxContainerSize);
                        }
                        var childrenCount = applySafetyLimits && count > WafConstants.MaxContainerSize ? WafConstants.MaxContainerSize : count;
```

I think that's why the PathParamsAttack unit test is failing, as the waf gets loaded with truncated rules (we apply limits only for inputs coming from the request but not at waf init)